src/ServiceControl.Config/UI/Shell/VersionCheckerHelper.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace ServiceControl.Config.UI.Shell
+{
+    using Newtonsoft.Json;
+    using ServiceControlInstaller.Engine.Configuration.ServiceControl;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+
+    public static class VersionCheckerHelper
+    {
+        public static async Task<Release> GetDownloadUrlForNextVersionToUpdate(string currentVersion)
+        {
+            Version current = new Version(currentVersion);
+            List<Release> releases = await GetVersionInformation().ConfigureAwait(false);
+
+            Release topversion = releases.Select(t => (t.Version, t)).Max().t;
+            
+            UpgradeInfo upgradeInfo = UpgradeControl.GetUpgradeInfoForTargetVersion(topversion.Version, current);
+            
+            Release recomendedVersion = releases.FirstOrDefault(t => t.Version == upgradeInfo.RecommendedUpgradeVersion);
+
+            if (recomendedVersion != null && recomendedVersion.Version > current)
+                return recomendedVersion;
+            // we have no release available
```

Comment:
This may not be true. If there's a recommended upgrade version then that version must exist, but it may not exist in the document found at the link provided which is limited to the last X releases. i.e. If you were running version 1.5 today, it would give you a recommended upgrade of 1.48, but that's not in the json file so we'd end up here.

Reply:
Is it good enough if we just show "New version available" that points to the download? If someone sees that they can decide to download the newest version that will have the update restrictions in place and will guide them when doing the instance level upgrades.

Reply:
Not sure what are we suggesting to do here...

Reply:
I would drop the reference to `UpgradeControl`. If there's an update in the list on the web, then we should tell the user about it.

Reply:
I've changed this to select the newest version if it's higher then that current one